staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -147,7 +147,7 @@ func NewCmdDiff(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.C
 			// return 1 if there was a problem.
 			if err := options.Run(); err != nil {
 				if exitErr := diffError(err); exitErr != nil {
-					os.Exit(exitErr.ExitStatus())
+					cmdutil.CheckErr(err)
```

Comment:
I see what you mean by wanting to use `CheckErr` so that you can override `BehaviorOnFatal` in your code. However, I think this specific change breaks the behaviour described in the comment above:

> we don't want to print an error if diff returns error code 1, which simply means that changes were found.

In other words, kubectl's default fatal error handler is not suitable for this case, so the standard flow through cmdutil.CheckErr can't be used.

Reply:
I understand what you mean. Would there be an approach you'd recommend to resolve this issue that I described in this situation?

Reply:
It looks like if we do `cmdutil.CheckErr(cmdutil.ErrExit)` instead of passing through the real error, we'll get the behaviour we want. From `helpers.go`:

```go
// ErrExit may be passed to CheckError to instruct it to output nothing but exit with
// status code 1.
var ErrExit = fmt.Errorf("exit")
```